File: 8kyu/merge-two-sorted-arrays-into-one/merge-two-sorted-arrays-into-one.go
package main

import (
	"fmt"
	"sort"
)

func Map(merged map[int]bool, arr []int) {
	for _, num := range arr {
		merged[num] = true
	}
}

func MergeArrays(arr1, arr2 []int) []int {
	merged := make(map[int]bool)

	Map(merged, arr1)
	Map(merged, arr2)

	res := make([]int, 0, len(merged))

	for key := range merged {
		res = append(res, key)
	}

	sort.Ints(res)

	return res
}

func main() {
	fmt.Println(MergeArrays([]int{1, 2, 3, 4}, []int{5, 6, 7, 8}))
}

File: 8kyu/merge-two-sorted-arrays-into-one/merge-two-sorted-arrays-into-one.py
# def merge_arrays(arr1, arr2):
#     res = list(set(arr1 + arr2))
#     res.sort()
#     return res

def merge_arrays(arr1, arr2):
    res = sorted(set(arr1 + arr2))
    return res

print(merge_arrays([1, 3, 5, 7], [2, 4, 6, 8]))
print(merge_arrays([1,3,5,7,9], [10,8,6,4,2]))
print(merge_arrays([1,3,5,7,9,11,12], [1,2,3,4,5,10,12]))